packages/core/src/browser/quick-open/quick-input-title-bar.ts

```diff
@@ -0,0 +1,234 @@
+/********************************************************************************
+ * Copyright (C) 2019 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { TitleButton } from './quick-open-service';
+import { Emitter } from '../../common/event';
+import { DisposableCollection } from '../../common/disposable';
+import { FOLDER_ICON, FILE_ICON } from '..';
+import { BuiltinThemeProvider } from '../theming';
+import URI from '../../common/uri';
+
+export class InputTitleBar {
```

Comment:
- please always use specific names, i.e. `QuickInputTitleBar`
- always name a class after the file or other way around

It's done to easy search and part fo Theia project coding guidelines: https://github.com/theia-ide/theia/wiki/Coding-Guidelines#names. The goal is to put a class name in the file search and see only one hit.